.circleci/config.yml

```diff
@@ -146,60 +148,76 @@ jobs:
       - run: bundle check --path vendor/bundle || bundle install --path vendor/bundle
+      - run:
+          name: Ensure Local Bin Drectory exists
+          command: mkdir -p ~/.local/bin
+      - run:
+          name: Setup MSOffice Crypt
+          command: |
+            if [ `which msoffice-crypt` ]
+            then
+              echo "msoffice-crypt cached ok"
+            else
+              mkdir ./build
+              cd ./build
+              git clone https://github.com/herumi/cybozulib
+              git clone https://github.com/herumi/msoffice
+              cd msoffice
+              make -j RELEASE=1
+              mv ./bin/msoffice-crypt.exe ~/.local/bin/msoffice-crypt
+              chmod +x ~/.local/bin/msoffice-crypt
+              cd ../..
+              rm -rf build
+            fi
+      - run:
+          name: Download Code Climate
+          command: |
+            if [ `which cc-test-reporter` ]
+            then
+              echo "CC test reporter cached"
+            else
+              curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ~/.local/bin/cc-test-reporter
+              chmod +x ~/.local/bin/cc-test-reporter
+            fi
       - save_cache:
           key: allocation-manager-v2-{{ checksum "Gemfile.lock" }}
           paths:
             - ~/repo/vendor/bundle
+            - ~/.local/bin
       - persist_to_workspace:
-          root: .
+          root: ~/
           paths:
-            - vendor/bundle
-            - node_modules/
+            - repo/vendor/bundle
+            - repo/node_modules/
+            - .local/bin
 
   test:
     <<: *defaults
     <<: *test_container_config
     steps:
       - checkout
       - attach_workspace:
-          at: ~/repo
+          at: ~/
       - run: bundle --path vendor/bundle
-      - run:
-          name: Migrate database
-          command: |
-            bundle exec rake db:create db:schema:load --trace db:migrate
       - run:
           name: Rubocop
           command: bundle exec rubocop
       - run:
           name: Security analysis
           command: bundle exec brakeman -o ~/test-results/brakeman/brakeman.json -o ~/test-results/brakeman/brakeman.html
       - run:
-          name: Download Code Climate
-          command: |
-            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
-            chmod +x ./cc-test-reporter
-      - run:
-          name: Setup MSOffice Crypt
+          name: Setup database
           command: |
-            mkdir ./build
-            cd ./build
-            git clone https://github.com/herumi/cybozulib
-            git clone https://github.com/herumi/msoffice
-            cd msoffice
-            make -j RELEASE=1
-            sudo mv ./bin/msoffice-crypt.exe /usr/local/bin/msoffice-crypt
-            cd ../..
-            rm -rf build
+            bundle exec rake db:setup[4]
```

Comment:
I've Googled it, but haven't seen anything obvious. What's the meaning of the `[4]` argument on the `db:setup` rake task?